core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -956,6 +980,16 @@ public IndexMetaData build() {
                 throw new IllegalArgumentException("must specify non-negative number of shards for index [" + index + "]");
             }
 
+            Integer maybeRoutingPartitionSize = settings.getAsInt(SETTING_ROUTING_PARTITION_SIZE, null);
```

Comment:
just use `INDEX_ROUTING_PARTITION_SIZE_SETTING.get(settings)` here and you will be fine. you also don't need the `routingPartitionSize <= 0 ` anymore since the setting has a min value of 1 afaik?